flink/v1.15/flink/src/test/java/org/apache/iceberg/flink/sink/TestFlinkIcebergSink.java

```diff
@@ -378,18 +371,14 @@ public void testOverrideWriteConfigWithUnknownFileFormat() {
             .tableLoader(tableLoader)
             .writeParallelism(parallelism)
             .setAll(newProps);
-
-    AssertHelpers.assertThrows(
-        "Should fail with invalid file format.",
-        IllegalArgumentException.class,
-        "Invalid file format: UNRECOGNIZED",
-        () -> {
-          builder.append();
-
-          // Execute the program.
-          env.execute("Test Iceberg DataStream.");
-          return null;
-        });
+    Assertions.assertThatThrownBy(
```

Comment:
same as above. the error happens when append is called, not when execute is called